File: linkedList/ll0.cpp
#include <iostream>
using namespace std;

class Node
{
public:
  int data;
  Node* next;

  Node(int n=0){
    this->data = n;
    this->next = NULL;
  }
};

void printLL(Node* head)
{

  if (head)
  {
    cout << head->data << endl;
    printLL(head->next);

  }
}

void printRev(Node* head)
{
  if (head)
  {
    printRev(head->next);
    cout << head->data << endl;
  }
}

int main()
{
  Node* head = new Node(10);
  head->next = new Node(20);
  head->next->next = new Node(30);
  printLL(head);
  printRev(head);
}

File: linkedList/ll0.py
class Node:
    def __init__(self, data=0):
        self.data = data
        self.next = None



def printLL(head):
    if head != None:
        print(head.data)
        printLL(head.next)



def printRev(head):
    if head != None:
        printRev(head.next)
        print(head.data)


head = Node(10)
head.next = Node(20)
head.next.next = Node(30)
printLL(head)
printRev(head)
